api/server/httputils/errors.go

```diff
@@ -72,6 +75,104 @@ func GetHTTPErrorStatusCode(err error) int {
 	return statusCode
 }
 
+// FromStatusCode creates an errdef error, based on the provided status-code
+func FromStatusCode(err error, statusCode int) error {
+	if err == nil {
+		return err
+	}
+	switch statusCode {
+	case http.StatusNotFound:
+		err = errdefs.NotFound(err)
+	case http.StatusBadRequest:
+		err = errdefs.InvalidParameter(err)
+	case http.StatusConflict:
+		err = errdefs.Conflict(err)
+	case http.StatusUnauthorized:
+		err = errdefs.Unauthorized(err)
+	case http.StatusServiceUnavailable:
+		err = errdefs.Unavailable(err)
+	case http.StatusForbidden:
+		err = errdefs.Forbidden(err)
+	case http.StatusNotModified:
```

Comment:
This may actually have to be discussed as well; this maps to a `304 Not Modified`, so not an "error" (similar to https://github.com/moby/moby/pull/38462);

https://github.com/moby/moby/blob/cb501886db2b46594a66cc2f54e81b53097e5fd8/errdefs/helpers.go#L117-L131

Reply:
`NotModified()` _is_ in active use (see https://github.com/moby/moby/blob/d4a6e1c44f4ce7a974a2142cf083188c07d85fbd/daemon/start.go#L36-L38). I guess the confusing bit there is that it's treated as an `error` but not really an `error`. Maybe just semantics 🤷‍♂️